sorc/chgres_cube.fd/input_data.F90

```diff
@@ -6607,4 +6603,33 @@ recursive subroutine quicksort(a, first, last)
   if (j+1 < last)  call quicksort(a, j+1, last)
 end subroutine quicksort
 
+!> Check for and replace erroneous values in soil temperature.
```

Comment:
I would add more detail here. You are not just updating soil temperature, but ice temperature as well. And for open water, you are using SST as a 'filler' value, since this array is not used at those points.

Reply:
That is the chgres_cube convention which can confuse people. The soil temperature variables only hold soil temperature at land points.